pkg/planner/property/physical_property.go

```diff
@@ -297,6 +305,75 @@ func (p *PhysicalProperty) IsSubsetOf(keys []*MPPPartitionColumn) []int {
 	return matches
 }
 
+// NeedEnforceExchangerWithHashByEquivalence checks if the keys can match the needs of partition with equivalence.
+// "Equivalence" refers to the process where we utilize a hash column to obtain equivalent columns,
+// and then use these equivalent columns to compare with the MPP partition column to determine whether an exchange is
+// necessary.
+//
+// for example:
+//  1. MPPPartitionCols: [18，13，16]
+//  2. keys: [9]
+//  3. FD: (1)-->(2-6,8), ()-->(7), (9)-->(10-17), (1,10)==(1,10), (18,21)-->(19,20,22-33), (9,18)==(9,18)
+//     In this case, we can see that the child supplied partition keys is subset of parent required partition cols.
+func (p *PhysicalProperty) NeedEnforceExchangerWithHashByEquivalence(keys []*MPPPartitionColumn) bool {
+	// keys is the HashCol. If the partition cols are a subset of the hash cols, then need to enforce exchange.
+	if len(p.MPPPartitionCols) < len(keys) {
+		return true
+	}
+	uniqueID2MppCol := make(map[*MPPPartitionColumn]intset.FastIntSet, len(keys))
```

Comment:
Please use len(MPPPartitionCols) instead

Reply:
it has been updated.